### PR TITLE
[REGEDIT] Correctly limit splitter's X coordinates

### PR DESCRIPTION
**Pull request by katahiromz**
## Purpose

JIRA issue: [CORE-19516](https://jira.reactos.org/browse/CORE-19516)

## Proposed changes

- Add `ClampSplitBarX` helper function to delete duplicated code.
- Use a signed value in `WM_LBUTTONUP` handling.
- Use `ClampSplitBarX` in `ResizeWnd` function, `WM_LBUTTONDOWN`, `WM_MOUSEMOVE`, and `WM_LBUTTONUP` handling.

## TODO

- [x] Do tests.

## Video

https://github.com/reactos/reactos/assets/2107452/d10a53ad-233e-44de-b41e-fec27e57e624

### CODE REVIEW

File: base/applications/regedit/childwnd.c
Comment (JoachimHenze) [inline base/applications/regedit/childwnd.c:478]: Is that is some ancient left-over of a timer that is no longer present?

Comment (katahiromz) [reply base/applications/regedit/childwnd.c:478]: There is no `SetTimer` call. It's meaningless.

```bash
$ git blame base/applications/regedit/childwnd.c
...
a0f2e6b8cacd rosapps/regedit/childwnd.c                   (Robert Dickenson      2002-07-14 21:03:10 +0000 476)     case WM_TIMER:
a0f2e6b8cacd rosapps/regedit/childwnd.c                   (Robert Dickenson      2002-07-14 21:03:10 +0000 477)         break;
...
```

**Review by JoachimHenze — APPROVED** [review]
Approved. Thank you for the prompt reaction. I see nothing wrong when reviewing formally. If it works, nice. I do recommend to wait for another devs review/approval though before merging.
When I reproduced the bug myself in unpatched state, I dragged leftmost, then released the mouse. And then immediately trying to move left-most again in the next drag-session (without moving it right-most in between). I am not sure whether that's the perfect reproduction steps to make it happen in unpatched state, but it was what worked for me.

I wasn't able to repro the bug with the very first drag already, or when dragging repeatedly and alternatively dragLeftmost-release-DragRightmost-release-DragLeftmost...

**Review by HBelusca — APPROVED** [review]

**Pull request merged**
